Fix adding shortcuts in Android 12.1+

proletarius101 opened the pull request:
Fix #342

Tested on Android Emulator with API 32.

proletarius101 committed:
Fix adding shortcuts in Android 12.1+

proletarius101 committed:
Remove unused imports

proletarius101 committed:
Remove unused variable

proletarius101 commented:
Ssince the APIs I use supports them according to the doc, I believe it works for all Island supported platform levels:
- https://developer.android.com/reference/androidx/core/content/pm/ShortcutManagerCompat#setDynamicShortcuts(android.content.Context,java.util.List%3Candroidx.core.content.pm.ShortcutInfoCompat%3E)
- https://developer.android.com/reference/androidx/core/content/pm/ShortcutManagerCompat#updateShortcuts(android.content.Context,java.util.List%3Candroidx.core.content.pm.ShortcutInfoCompat%3E)

proletarius101 committed:
Fix sdk version matching

proletarius101 committed:
Fix adding shortcuts in Android 12.1+

Hima88888888 reviewed (commented):
- []()

proletarius101 commented:
You may check https://cloudflare.f-droid.org/en/packages/com.oasisfeng.island.fdroid/ directly for the fixed usable version

oasisfeng commented:
This pull request removed the following code in the implementation of Island app-launch shortcut:
```
if (IslandManager.isReady(context, profile))
    Shuttle(context, profile).launchNoThrows { requestPinAsUser(this, app, dynamic) }
```
This piece of code gives the shortcut different badge for different Island.

Is that the cause of the mentioned issue?

proletarius101 committed:
fix: `Max number of dynamic shortcuts exceeded`

use Google Shortcuts Integration Library (https://developer.android.com/develop/ui/views/launch/shortcuts/creating-shortcuts#gsi-library) to bypass the maximum shortcut limit

proletarius101 commented:
Hi @oasisfeng , can you explain why do you need to add dynamic shortcuts? The instruction in https://developer.android.com/develop/ui/views/launch/shortcuts/creating-shortcuts shows that only pinned shortcuts are useful in this case. Or is that useful when you enable the frozen icon indicator feature? If that's the case, it's a misunderstanding of dynamic shortcuts. Pinned icons labels can also be changed. 
That helps my investigation

Av-Srt reviewed (approved):
#347

AtFreezingPoint reviewed (approved):
Looks good
